internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -295,6 +296,39 @@ func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest) (string, *error
 	return resp.Subscription().ID(), nil
 }
 
+func (q amsQuotaService) ReserveQuotaIfNotAlreadyReserved(kafka *dbapi.KafkaRequest) (string, *errors.ServiceError) {
+	instanceType, err := q.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(kafka.InstanceType)
+	if err != nil {
+		return "", errors.GeneralError("failed checking current quota: %v", err)
+	}
+	kafkaBillingModel, err := instanceType.GetKafkaSupportedBillingModelByID(kafka.DesiredKafkaBillingModel)
+	if err != nil {
+		return "", errors.GeneralError("failed checking current quota: %v", err)
+	}
+
+	subscriptions, err := q.amsClient.FindSubscriptions(fmt.Sprintf("cluster_id='%s'", kafka.ClusterID))
```

Comment:
Can't we check the subscription_id? this would require storing it in the DB in this step itself and not at the end :thinking:

Reply:
Nope. This method is used by the reconciler so that it can recover even in case of errors updating the database.